src/main/java/org/springframework/data/couchbase/config/AbstractCouchbaseConfiguration.java

```diff
@@ -145,14 +146,14 @@ protected void configureEnvironment(final ClusterEnvironment.Builder builder) {
 
 	@Bean(name = BeanNames.COUCHBASE_TEMPLATE)
 	public CouchbaseTemplate couchbaseTemplate(CouchbaseClientFactory couchbaseClientFactory,
-			MappingCouchbaseConverter mappingCouchbaseConverter) {
-		return new CouchbaseTemplate(couchbaseClientFactory, mappingCouchbaseConverter);
+			MappingCouchbaseConverter mappingCouchbaseConverter, TranslationService translationService) {
```

Comment:
this breaks backwards compatibility by introducing another required argument - if this should be added we need another overload that keeps the defaults as is I think

Reply:
Hi @daschl 

You are right about the overloading of the _CouchbaseTemplate_ and _ReactiveCouchbaseTemplate_ constructors. I'll update it now.

As for the definition of beans _couchbaseTemplate_ and _reactiveCouchbaseTemplate_, do you consider any change necessary? I didn't quite understand you.

Reply:
Yeah, I was actually also talking about the bean definitions - because if someone overloads them in their actual configuration, just changing the signature on update will fail to compile for them.

Reply:
Hi @daschl 

It's true that is a breaking change, but I can't think of any other way to do it. Bean _couchbaseTranslateService_ is being created but then not used.


If you don't want to be breakers, we could keep the overload on the _CouchbaseTemplate_ and _ReactiveCouchbaseTemplate_ constructors and at another time add the TranslationService parameter to the configuration beans and then be breakers.

What do you think?

Thank you very much

Reply:
@jorgerod well, you could make the beans take the new parameter by default (as you do now), but add another 2 methods that look like the old signature and not be a Bean, which call into the new beans and just initialize the json translation service. Then if someone overrides it, their code would not break but everyone can then switch over to the new bean if needed?

Reply:
I understand. Thank you very much for the help  @daschl